### PR TITLE
Support tiered locality scheduling for app pod

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
Support app pod scheduling with tiered locality affinity, 
Design doc, see https://www.yuque.com/xliuqq-bgkty/nxcpc4/hvkeaef22quuw6pe

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
unit test

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: api/v1alpha1/status.go [hunk 1]
```diff
@@ -128,6 +128,9 @@ type RuntimeStatus struct {
 
 	// MountPoints represents the mount points specified in the bounded dataset
 	Mounts []Mount `json:"mounts,omitempty"`
+
+	// WorkerNodeAffinity represents the runtime worker pods node affinity including node selector
+	WorkerNodeAffinity *corev1.NodeAffinity `json:"workNodeAffinity,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/status.go:133]: How about using CacheAffinity as name?

Comment (xliuqq) [reply api/v1alpha1/status.go]: ok, fixed.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**